processing/src/main/java/io/druid/query/monomorphicprocessing/SpecializationService.java

```diff
@@ -124,7 +125,7 @@ protected PerPrototypeClassState computeValue(Class<?> type)
       String runtimeShape
   )
   {
-    return getSpecializationState(prototypeClass, runtimeShape, ImmutableMap.<Class<?>, Class<?>>of());
+    return getSpecializationState(prototypeClass, runtimeShape, ImmutableMap.of());
```

Comment:
> is synchronization required on call to perPrototypeClassState.get() from multiple threads? or call to computeValue()?

According to `ClassValue` documentation, it should be safe: "The actual installation of the value on the class is performed atomically. At that point, if several racing threads have computed values, one is chosen, and returned to all the racing threads."